ko/publish/daemon.go

```diff
@@ -34,7 +35,11 @@ func NewDaemon(wo daemon.WriteOptions) Interface {
 
 // Publish implements publish.Interface
 func (d *demon) Publish(img v1.Image, s string) (name.Reference, error) {
-	tag, err := name.NewTag(s, name.WeakValidation)
+	h, err := img.Digest()
+	if err != nil {
+		return nil, err
+	}
+	tag, err := name.NewTag(fmt.Sprintf("local.dev/%s:%s", s, h.Hex), name.WeakValidation)
```

Comment:
What about `ko.local` instead of `local.dev`?

https://tools.ietf.org/html/rfc6762